src/VapStuff/modules/workbenches/SimpleAlchemyStation/recipes/MagmaticSponge.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable max-len */
+import ItemBuilder from '../../../../../lib/com/smc/utils/ItemBuilder.js'
+import Player from '../../../../../lib/org/bukkit/entity/Player.js'
+import BlockPlaceEvent from '../../../../../lib/org/bukkit/event/block/BlockPlaceEvent.js'
+import GameMode from '../../../../../lib/org/bukkit/GameMode.js'
+import ItemStack from '../../../../../lib/org/bukkit/inventory/ItemStack.js'
+import Material from '../../../../../lib/org/bukkit/Material.js'
+import Sound from '../../../../../lib/org/bukkit/Sound.js'
+import { Subscribe } from '../../../../services/EventListener.js'
+import Module from '../../../../types/Module.js'
+import { colorText, getDeltasFromTaxicabRadius } from '../../../../util.js'
+
+interface MagmaticSpongeData {
+  isLavaWet: boolean;
+}
+
+@Subscribe
+export default class MagmaticSponge extends Module {
+  get name() { return 'Magmatic Sponge' }
+
+  readonly ID = 'magmaticSponge'
+
+  readonly SPONGE_NAME = colorText('&#FF3300Magmatic Sponge')
+
+  readonly SPONGE_NAME_LAVAWET = colorText('&#FF3300Lavawet Magmatic Sponge')
+
+  readonly RADIUS = 7
+
+  onEnable() {
+    this.plugin.extraRecipes.addRecipe('simpleAlch', this.ID, {
+      waterUsage: 3,
+      name: 'Magmatic Sponge',
+      ingredients: [
+        [Material.SPONGE, 1],
+        [Material.MAGMA_BLOCK, 4],
+        [Material.MAGMA_CREAM, 4],
+        [Material.BLAZE_POWDER, 1],
+      ],
+      createResult: () => this.createMagmaticSponge(),
+    })
+    this.plugin.blockAspects.addAspect(this.ID, {
+      serializeItem: (item: ItemStack): MagmaticSpongeData => {
+        if (item.getType() === Material.WET_SPONGE && this.isMagmaticSpongeLavaWet(item)) {
+          return { isLavaWet: true }
+        }
+        if (item.getType() === Material.SPONGE && this.isMagmaticSponge(item)) {
+          return { isLavaWet: false }
+        }
+        return null
+      },
+      onPlace: (data: MagmaticSpongeData, event: BlockPlaceEvent) => {
+        const location = event.getBlockPlaced().getLocation()
+        const locationDataType = location.getBlock().getType()
+        const blocks = getDeltasFromTaxicabRadius(this.RADIUS)
+        const player = event.getPlayer()
+        for (const block of blocks) {
+          const deltaLoc = location.clone().add(...block)
+          if (deltaLoc.getBlock().getType() === Material.WATER && locationDataType === Material.SPONGE) {
+            event.setCancelled(true)
+            player
+              .getWorld()
+              .playSound(player.getLocation(), Sound.BLOCK_LAVA_EXTINGUISH, 0.5, 0.5)
```

Comment:
The last two parameters of `World#playSound` are `volume` and `pitch` respectively. I'm not sure if you meant to half the pitch, but if this is still too loud then you should lower `volume`. Keep in mind that in JVM this method takes a Float, so GraalVM might take issue with casting the Double into a Float. If needed, you can convert your number into a "float-safe" number with `Module#floatSafe`.